tools/walletextension/rpcapi/utils.go

```diff
@@ -149,6 +154,8 @@ func ExecAuthRPC[R any](ctx context.Context, w *Services, cfg *ExecCfg, method s
 		return nil, rpcErr
 	})
 
+	w.RateLimiter.SetRequestEnd(gethcommon.Address(userID), requestUUID)
```

Comment:
one comment. I think this should better be in a defer 
(Same for filter_api)

Reply:
fixed.